yb-voyager/cmd/analyzeSchema.go

```diff
@@ -593,6 +593,72 @@ func checker(sqlInfoArr []sqlInfo, fpath string, objType string) {
 	}
 }
 
+func checkRedundantIndexes() {
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("error getting migration status record: %v", err)
+	}
+	if !msr.MigrationAssessmentDone { //TODO: add the case to handle if assessment is run in export schema
+		return
+	}
+
+	if msr.SourceDBConf.DBType != POSTGRESQL {
+		return
+	}
+
+	//Only considering the case where assessment is run in this migration
+	//If not via assess-migration, there is going to be a change where internally in the export-schema we will run assessment
+	//so we should have this information most of the time.
+	assessmentReportPath := filepath.Join(exportDir, "assessment", "reports", fmt.Sprintf("%s.json", ASSESSMENT_FILE_NAME))
+	if !utils.FileOrFolderExists(assessmentReportPath) {
+		log.Infof("migration assessment report file doesn't exists at %q, skipping apply recommendations step...", assessmentReportPath)
+		return
+	}
+
+	log.Infof("parsing assessment report json file for redundant indexes")
+	report, err := ParseJSONToAssessmentReport(assessmentReportPath)
+	if err != nil {
+		utils.ErrExit("error parsing json report file %q: %w", assessmentReportPath, err)
+	}
+	assessmentIssues := report.Issues
+	redundantIndexIssues := lo.Filter(assessmentIssues, func(i AssessmentIssue, _ int) bool {
+		return i.Category == PERFORMANCE_OPTIMIZATIONS_CATEGORY && i.Name == queryissue.REDUNDANT_INDEXES_ISSUE_NAME
+	})
+	for _, redundantIndexIssue := range redundantIndexIssues {
+		//this is the Unquoted object name
```

Comment:
Here's an alternative that comes to mind:

Instead of reading from the report, read from the assessmentDB and create issues out of that. 
Basically, exactly what you're doing in `addAssessmentIssuesForRedundantIndexes` except you return analyzeIssues instead of assessmentIssues. 

Will that simplify things?

Reply:
also think about upgrade scenarios.

Reply:
I thought about this, but there is a case where assessment is run with a `metadata dir` where the user can use the script and get the information, and provide it. The assessment will generate the report, so for such we will need the `metadata-dir-path`, but that is not possible to get and will create problems in Docker even if we store that in MSR to use here. 
So, I thought in all cases we will have the report with and its in export-dir so shouldn't be a problem in any environment.

Reply:
> also think about upgrade scenarios.

In upgrade, if the  assessment is run withan  older version and analyze with a newer one, we will not have `redundantIndexes` issues filtered from assessment issues so we will not report them, which is okay I think?

Reply:
hmm, makes sense. Can you put this as a comment in this function as to why we chose this approach?